### ISSUE TITLE
Found RM's out_features is 2, when trying to use reward model to do ppo

### ISSUE THREAD

**Issue by jijivski** [OpenRLHF/OpenRLHF#1039]
```python
import torch
from transformers import AutoModelForSequenceClassification
opnerlhf_rm_path = 'OpenRLHF/Llama-3-8b-rm-700k'
opnerlhf_model = AutoModelForSequenceClassification.from_pretrained(opnerlhf_rm_path)
# (Pdb) opnerlhf_model
# ...
#     (rotary_emb): LlamaRotaryEmbedding()
#   )
#   (score): Linear(in_features=4096, out_features=2, bias=False)
# )
```
I think this should be 1, and in OPENRLHF repo, the head is also set to be 1:
[openrlhf/models/model.py](https://github.com/OpenRLHF/OpenRLHF/blob/494850f50342ed38d5ae76ef45a3207f3523b582/openrlhf/models/model.py#L157)

```python 
def _get_reward_model(base_pretrained_model, base_llm_model, value_head_prefix="score", packing_samples=False):
    class RewardModel(base_pretrained_model):
        supports_gradient_checkpointing = True

        def __init__(self, config: AutoConfig):
            super().__init__(config)
            setattr(self, self.base_model_prefix, base_llm_model(config))

            self.value_head_prefix = value_head_prefix
            setattr(self, value_head_prefix, nn.Linear(config.hidden_size, 1, bias=False))
```
Thanks for any clue!

**Comment by jijivski** [comment]
```python
reward_model = AutoModelForSequenceClassification.from_pretrained(  
    reward_model_path,  
    num_labels=1,  
    torch_dtype=torch.bfloat16,  
    attn_implementation="flash_attention_2",  
    use_cache=False,  
)
```
OH, I found that all cases in OpenRLHF use num_labels=1,   to avoid default num_labels=2, in verl or other places you can do so to prevent warning.
